src/main/java/com/everis/d4i/tutorial/services/impl/TvShowServiceImpl.java

```diff
@@ -26,43 +28,112 @@
 
 	@Autowired
 	private TvShowRepository tvShowRepository;
+	@Autowired
+	private CategoryRepository categoryRepository;
+	@Autowired
+	private ChapterRepository chapterRepository;
 
 	private ModelMapper modelMapper = new ModelMapper();
 
-	/*@Override
+	private static final int FIRST_POSITION = 0;
+
+	@Override
 	public List<TvShowRest> getTvShowsByCategory(Long categoryId) throws NetflixException {
 
-		return tvShowRepository.findByCategoryId(categoryId).stream()
+		return tvShowRepository.findByCategoriesId(categoryId).stream()
 				.map(tvShow -> modelMapper.map(tvShow, TvShowRest.class)).collect(Collectors.toList());
 
-	}*/
+	}
 
 	@Override
 	public TvShowRest getTvShowById(Long id) throws NetflixException {
-		
-		TvShow show =  tvShowRepository.findById(id)
-				.orElseThrow(() -> new NotFoundException(ExceptionConstants.MESSAGE_INEXISTENT_SHOW));
-		
+
+		TvShow show = getById(id);
+
 		return modelMapper.map(show, TvShowRest.class);
 	}
-	
+
 	@Override
-	public List<TvShowRest> getShowAwards() throws NetflixException {
+	public TvShowRest updateName(String name, Long id) throws NetflixException {
+		TvShow show = getById(id);
+
+		show.setName(name);
+
+		return modelMapper.map(saveTvShow(show), TvShowRest.class);
+	}
+
+	@Override
+	public TvShowRest addCategory(Long tvShowId, Long categoryId) throws NetflixException {
+		TvShow show = getById(tvShowId);
+		Category category = categoryRepository.findById(categoryId)
+				.orElseThrow(() -> new NotFoundException(ExceptionConstants.MESSAGE_INEXISTENT_CATEGORY));
+
+		List<Category> categoriesList = show.getCategories();
+		categoriesList.add(category);
+		show.setCategories(categoriesList);
+
+		List<TvShow> showList = category.getTvShows();
+		showList.add(show);
+		category.setTvShows(showList);
+		categoryRepository.saveAndFlush(category);
+
+		return modelMapper.map(saveTvShow(show), TvShowRest.class);
```

Comment:
you are presisting the same relationship 2 times and flushing. Be care about this and try to do this in the same transaction